components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/template/ThrottlePolicyTemplateBuilder.java

```diff
@@ -0,0 +1,411 @@
+        policyTemplateLocation = path;
+    }
+
+    /**
+     * Generate policy for api level throttling
+     *
+     * @param policy Policy with level 'api'. isAcrossAllUsers() method in policy is used to identify the level in
+     *               the api level. Policy can have multiple pipelines and a default condition which will be used as
+     *               else condition
+     * @return
+     * @throws APITemplateException
+     */
+    public Map<String, String> getThrottlePolicyForAPILevel(APIPolicy policy) throws APITemplateException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Generating policy for apiLevel :" + policy.toString());
+        }
+        Map<String, String> policyArray = new HashMap<String, String>();
+        VelocityEngine velocityengine = new VelocityEngine();
+        if (!"not-defined".equalsIgnoreCase(getVelocityLogger())) {
+            velocityengine.setProperty(RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+                    "org.apache.velocity.runtime.log.Log4JLogChute");
+            velocityengine.setProperty("runtime.log.logsystem.log4j.logger", getVelocityLogger());
+        }
+        velocityengine.init();
+        Template template = velocityengine.getTemplate(getTemplatePathForAPI());
+        StringWriter writer;
+        VelocityContext context;
+
+        if (policy.getPipelines() != null) {
+
+            for (Pipeline pipeline : policy.getPipelines()) {
+                context = new VelocityContext();
+                setConstantContext(context);
+                context.put("pipelineItem", pipeline);
+                context.put("policy", policy);
+
+                context.put("quotaPolicy", pipeline.getQuotaPolicy());
+                context.put("pipeline", "condition_" + pipeline.getId());
+
+                String conditionString = getPolicyCondition(pipeline.getConditions());
+                //                    conditionsSet.add(conditionString);
+
+                context.put("condition", " AND " + conditionString);
+                writer = new StringWriter();
+                template.merge(context, writer);
+                if (log.isDebugEnabled()) {
+                    log.debug("Policy : " + writer.toString());
+                }
+
+                String policyName =
+                        PolicyConstants.POLICY_LEVEL_RESOURCE + "_" + policy.getPolicyName() + "_condition_" + pipeline
+                                .getId();
+                policyArray.put(policyName, writer.toString());
+            }
+        }
+
+        return policyArray;
+    }
+
+    /**
+     * Generate default policy for api level throttling
+     *
+     * @param policy Policy with level 'api'. isAcrossAllUsers() method in policy is used to identify the level in
+     *               the api level. Policy can have multiple pipelines and a default condition which will be used as
+     *               else condition
+     * @return
+     * @throws APITemplateException
```

Comment:
Add the description for APITemplateException in comments? Otherwise there will be check-style plugin warning when compiling. There are multiple places in the PR.